src/arm/mach/init.c

```diff
@@ -54,6 +60,54 @@ struct cpuinfo_arm_isa cpuinfo_isa = {
 #endif
 };
 
+struct cache_array {
```

Comment:
This looks redundant, just pass `struct cpuinfo_cache*` and `uint32_t count` as two separate variables.